nbgrader/apps/extensionapp.py

```diff
@@ -1,299 +1,35 @@
-            self.log.info("Activating assignment_list server extension")
-            self.enable_server_extension('nbgrader.nbextensions.assignment_list')
-
-            self.section = "tree"
-            self.toggle_nbextension("assignment_list/main")
-
-        self.log.info("Done. You may need to restart the Jupyter notebook server for changes to take effect.")
-
-
-class ExtensionDeactivateApp(DisableNBExtensionApp, NbGrader):
-
-    name = u'nbgrader-extension-deactivate'
-    description = u'Deactivate the nbgrader extension'
-
-    flags = {}
-    aliases = {}
-
-    examples = """
-
-    To deactivate all the nbgrader extensions:
-
-        nbgrader extension deactivate
-
-    To deactivate only a specific extension, you can pass the name of the
-    extension you want to deactivate as an argument, e.g.:
-
-        nbgrader extension deactivate create_assignment
-        nbgrader extension deactivate assignment_list
-
-    """
-
-    def _classes_default(self):
-        return [ExtensionDeactivateApp, DisableNBExtensionApp]
-
-    def _recursive_get(self, obj, key_list):
-        if obj is None or len(key_list) == 0:
-            return obj
-        return self._recursive_get(obj.get(key_list[0], None), key_list[1:])
-
-    def disable_server_extension(self, extension):
-        loader = JSONFileConfigLoader('jupyter_notebook_config.json', jupyter_config_dir())
-        try:
-            config = loader.load_config()
-        except ConfigFileNotFound:
-            config = Config()
-
-        if 'server_extensions' not in config.NotebookApp:
-            return
-        if extension not in config.NotebookApp.server_extensions:
-            return
-
-        config.NotebookApp.server_extensions.remove(extension)
-
-        # save the updated config
-        with io.open(os.path.join(jupyter_config_dir(), 'jupyter_notebook_config.json'), 'w+') as f:
-            f.write(six.u(json.dumps(config, indent=2)))
-
-    def start(self):
-        if len(self.extra_args) == 0 or "create_assignment" in self.extra_args:
-            self.log.info("Deactivating create_assignment nbextension")
-            self.section = "notebook"
-            self.toggle_nbextension("create_assignment/main")
-
-        if sys.platform != 'win32' and (len(self.extra_args) == 0 or "assignment_list" in self.extra_args):
-            self.log.info("Deactivating assignment_list server extension")
-            self.disable_server_extension('nbgrader.nbextensions.assignment_list')
-
-            self.log.info("Deactivating assignment_list nbextension")
-            self.section = "tree"
-            self.toggle_nbextension("assignment_list/main")
-
-        self.log.info("Done. You may need to restart the Jupyter notebook server for changes to take effect.")
+    $ jupyter serverextension enable --sys-prefix --py nbgrader
 
+To install for all users, replace `--sys-prefix` by `--system`.
```

Comment:
Good question! Right now the defaults for serverextensions and nbextensions are *different*, which we consider to be a bug. However fixing that bug is an API change so it is getting into the code base slowly. Because of this I *always* give one of the following explicitely: `--sys-prefix`, `--system` or `--user`.